skyvern/forge/sdk/workflow/service.py

```
@@ -869,287 +872,514 @@ async def _execute_workflow_blocks(
+                    indented_signature = textwrap.indent(normalized_signature, "        ")
+                    wrapper_code = f"async def __run_signature_wrapper():\n    return (\n{indented_signature}\n    )\n"
+
+                    LOG.debug("Executing run_signature wrapper", wrapper_code=wrapper_code)
 
-                    if not block.continue_on_failure:
-                        failure_reason = f"{block.block_type} block timed out. Reason: {block_result.failure_reason}"
-                        workflow_run = await self.mark_workflow_run_as_failed(
-                            workflow_run_id=workflow_run_id, failure_reason=failure_reason
+                    try:
+                        exec_code = compile(wrapper_code, "<run_signature>", "exec")
+                        exec(exec_code, exec_globals)
```

Ruff (0.14.7)

Use of `exec` detected

(S102)